src/network.c

```diff
@@ -585,22 +555,22 @@ int recv_any_msg(Connection* conn, int* type, void* msg, int* len,
  * @param socketfd The socket
  * @param buf The data
  * @param amount The data size
- * @return The number of bytes written, 0 on fatal error, and -1 on recoverable
- *         error.
+ * @return The number of bytes written, -1 on fatal error, and 0 on recoverable
+ *         error, just like write().
  */
 int write_raw(int socketfd, const char* buf, int amount) {
   int n;
   n = write(socketfd, buf, amount);
-  if (n == -1) {
+  if (n == 0) {
```

Comment:
I'm confused here.

From the write man page "On error, -1 is returned, and errno is set appropriately." And, "If a write() is interrupted by a signal handler before any bytes are written, then the call fails with the error EINTR; if it is interrupted after at least one byte has been written, the call succeeds, and returns the number of bytes written."

I don't think n will be zero if errno == EINTR. It will either be -1 or > 0. What is your interpretation?